src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Settings/ClientSettingsManager.cs

```diff
@@ -6,10 +6,12 @@
 using System.ComponentModel.Composition;
 using Microsoft.CodeAnalysis.ExternalAccess.Razor;
 using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Settings;
 using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.VisualStudio.Razor.Settings;
 
+[Export(typeof(IClientSettingsReader))]
```

Comment:
I'm confused that this actually works. I see that IClientSettingsReader has `GetClientSettings()` which this also has. Is that enough for MEF? It doesn't actually have to have the `IClientSettingsReader` in it's type ancestry?

Reply:
I made `IClientSettingsManager` implement `IClientSettingsReader` just so I didn't have to go and update everywhere that used it, so yes, it is in its type ancestry.